service/interpreter/stateExecutionCounter.go

```diff
@@ -114,10 +168,7 @@ func (e *StateExecutionCounter) updateStateIdSearchAttribute() error {
 	for sid := range e.stateIdCurrentlyExecutingCounts {
 		executingStateIds = append(executingStateIds, sid)
 	}
-	config := e.configer.Get()
-	if config.GetDisableSystemSearchAttribute() {
-		return nil
-	}
```

Comment:
this took me a few minutes to realize that's it's not breaking determinsm! you are right, and I agreed that moving this check to upper makes it easier to read.